hazelcast/src/main/java/com/hazelcast/topic/impl/reliable/MessageRunner.java

```diff
@@ -101,7 +101,7 @@ public void accept(ReadResultSet<ReliableTopicMessage> result, Throwable throwab
             // but we'll process whatever was received in 1 go.
 
             long lostCount = result.getNextSequenceToReadFrom() - result.readCount() - sequence;
-            if (lostCount > 0 && !isLossTolerable(lostCount)) {
+            if (lostCount != 0 && !isLossTolerable(lostCount)) {
```

Comment:
how can the loss be minus? when the cluster restarts?

Reply:
Yes, for example in `ClientReliableTopicOnClusterRestartTest.shouldFail_OnClusterRestart_whenDataLoss_notLossTolerant`. This test was failing before my last commit.

Reply:
Not sure if we should indeed cancel in this situation though...

Reply:
I'm ok with failing this case as well. Ideally, we would like to fail readers even if the new ringbuffer gets filled enough to catch up with the requested sequence see [here](https://github.com/hazelcast/hazelcast/issues/11857#issuecomment-352401655). 
This here is a best-effort approach for now.